persistence.js

```diff
@@ -102,64 +114,37 @@ function decodeRetainedPacket (chunk, enc, cb) {
   cb(null, msgpack.decode(chunk))
 }
 
-function Sub (clientId, topic, qos) {
-  this.clientId = clientId
-  this.topic = topic
-  this.qos = qos
-}
-
 RedisPersistence.prototype.addSubscriptions = function (client, subs, cb) {
   if (!this.ready) {
     this.once('ready', this.addSubscriptions.bind(this, client, subs, cb))
     return
   }
 
-  var multi = this._db.multi()
-
-  var clientSubKey = 'client:sub:' + client.id
-  var that = this
+  var clientSubKey = clientKey + client.id
 
   var toStore = {}
-
+  var offlines = []
   var count = 0
   var published = 0
   var errored = null
 
-  multi.exists(clientSubKey)
-
   for (var i = 0; i < subs.length; i++) {
     var sub = subs[i]
     toStore[sub.topic] = sub.qos
-
     if (sub.qos > 0) {
-      var subClientKey = 'sub:client:' + sub.topic
-      var encoded = msgpack.encode(new Sub(client.id, sub.topic, sub.qos))
-      multi.rpush(subscriptionsKey, subClientKey)
-      multi.hset(subClientKey, client.id, encoded)
+      offlines.push(sub.topic)
       count++
-      that._waitFor(client, sub.topic, finish)
+      // TODO don't wait the client an extra tick
```

Comment:
1. this is needed to provide a stable unit testing base. Mosca had so many spurious unit tests, and some of them relies on `setTimeout` to work. One of the underlining goal of aedes is to make maintenance and evolution simpler, and this goes into that direction.

2. no

I'm ok if you would like to try and remove that (or maybe disable it with an option?), but let's drop the TODO for now.